tools/osx/crosstool/osx_archs.bzl

```diff
@@ -26,32 +26,10 @@ OSX_TOOLS_NON_DEVICE_ARCHS = [
 ]
 
 OSX_TOOLS_ARCHS = [
-    "armeabi-v7a",
     "ios_armv7",
     "ios_arm64",
     "ios_arm64e",
     "watchos_armv7k",
     "watchos_arm64_32",
     "tvos_arm64",
 ] + OSX_TOOLS_NON_DEVICE_ARCHS
-
-# Target constraints for each arch.
-# TODO(apple-rules): Rename osx constraint to macOS.
-# TODO(apple-rules): Add constraints for watchos and tvos.
-OSX_TOOLS_CONSTRAINTS = {
-    "darwin_x86_64": ["@platforms//os:osx", "@platforms//cpu:x86_64"],
-    "darwin_arm64": ["@platforms//os:osx", "@platforms//cpu:aarch64"],
-    "darwin_arm64e": ["@platforms//os:osx", "@platforms//cpu:aarch64"],
-    "ios_i386": ["@platforms//os:ios", "@platforms//cpu:x86_32"],
-    "ios_x86_64": ["@platforms//os:ios", "@platforms//cpu:x86_64"],
-    "watchos_i386": ["@platforms//os:ios", "@platforms//cpu:x86_32"],
-    "watchos_x86_64": ["@platforms//os:ios", "@platforms//cpu:x86_64"],
-    "tvos_x86_64": ["@platforms//os:ios", "@platforms//cpu:x86_64"],
-    "armeabi-v7a": ["@platforms//cpu:arm"],
-    "ios_armv7": ["@platforms//os:ios", "@platforms//cpu:arm"],
-    "ios_arm64": ["@platforms//os:ios", "@platforms//cpu:aarch64"],
-    "ios_arm64e": ["@platforms//os:ios", "@platforms//cpu:aarch64"],
-    "watchos_armv7k": ["@platforms//os:ios", "@platforms//cpu:arm"],
-    "watchos_arm64_32": ["@platforms//os:ios", "@platforms//cpu:aarch64"],
-    "tvos_arm64": ["@platforms//os:ios", "@platforms//cpu:aarch64"],
-}
```

Comment:
This was only used in 1 file so I moved it there since it also slightly diffs now in keys vs the archs lists